chia/full_node/weight_proof.py

```diff
@@ -473,6 +475,8 @@ async def __slot_end_vdf(
             tmp_sub_slots_data.append(self.handle_block_vdfs(curr, blocks))
 
             curr = header_blocks[self.blockchain.height_to_hash(uint32(curr.height + 1))]
+            curr_header_hash = curr.header_hash
```

Comment:
Naive question: Do you really get a gain if this is set here at the end of the while loop nonetheless?
It also appears that if optimization is indeed a gain here, one could compute the two eos_vdf_iters's upfront.

Reply:
header_hash is a property it calls a function that streams the foliage blocks and then hashes it thats why it helps

Reply:
I think a better name for it would be `compute_hash()`, and for it to not look like a data member